parser.js

```diff
@@ -8,189 +8,29 @@
-         * specifically
-         */
-        if (additionalParsingContext.isParseForESLint) {
-            extra.parseForESLint = true;
-        }
-    }
-
-    if (!isRunningSupportedTypeScriptVersion && !warnedAboutTSVersion) {
-        const border = "=============";
-        const versionWarning = [
-            border,
-            "WARNING: You are currently running a version of TypeScript which is not officially supported by typescript-eslint-parser.",
-            "You may find that it works just fine, or you may not.",
-            `SUPPORTED TYPESCRIPT VERSIONS: ${SUPPORTED_TYPESCRIPT_VERSIONS}`,
-            `YOUR TYPESCRIPT VERSION: ${ACTIVE_TYPESCRIPT_VERSION}`,
-            "Please only submit bug reports when using the officially supported version.",
-            border
-        ];
-        extra.log(versionWarning.join("\n\n"));
-        warnedAboutTSVersion = true;
-    }
-
-    // Even if jsx option is set in typescript compiler, filename still has to
-    // contain .tsx file extension
-    const FILENAME = (extra.ecmaFeatures.jsx) ? "eslint.tsx" : "eslint.ts";
-
-    const compilerHost = {
-        fileExists() {
-            return true;
-        },
-        getCanonicalFileName() {
-            return FILENAME;
-        },
-        getCurrentDirectory() {
-            return "";
-        },
-        getDefaultLibFileName() {
-            return "lib.d.ts";
-        },
-
-        // TODO: Support Windows CRLF
-        getNewLine() {
-            return "\n";
-        },
-        getSourceFile(filename) {
-            return ts.createSourceFile(filename, code, ts.ScriptTarget.Latest, true);
-        },
-        readFile() {
-            return null;
-        },
-        useCaseSensitiveFileNames() {
-            return true;
-        },
-        writeFile() {
-            return null;
-        }
-    };
-
-    const program = ts.createProgram([FILENAME], {
-        noResolve: true,
-        target: ts.ScriptTarget.Latest,
-        jsx: extra.ecmaFeatures.jsx ? "preserve" : undefined
-    }, compilerHost);
-
-    const ast = program.getSourceFile(FILENAME);
-
-    extra.code = code;
-    return convert(ast, extra);
-}
+const parse = require("typescript-estree").parse;
+const astNodeTypes = require("typescript-estree").AST_NODE_TYPES;
+/* eslint-disable */
+const traverser = require("eslint/lib/util/traverser");
+/* eslint-enable */
 
 //------------------------------------------------------------------------------
 // Public
 //------------------------------------------------------------------------------
 
 exports.version = require("./package.json").version;
 
-exports.parse = function parse(code, options) {
-    return generateAST(code, options, { isParseForESLint: false });
-};
-
 exports.parseForESLint = function parseForESLint(code, options) {
-    const ast = generateAST(code, options, { isParseForESLint: true });
+    const ast = parse(code, options);
+    traverser.traverse(ast, {
```

Comment:
Can you refresh my memory again about when a function can exist without a body - is that only for type definitions?

Either way, if this shouldn't belong in the parser, I think this is a fine solution for now :+1:

Reply:
Here are some examples:

```ts
declare function f1(x: number): void;

interface Foo {
    bar(): boolean;
}
```